scripts/calculateRevenue.ts

```diff
@@ -23,13 +26,21 @@ async function main(args: ReturnType<typeof parseArgs>) {
   }> = []
 
   for (let i = 0; i < eligibleUsers.length; i++) {
-    const { referrerId, userAddress } = eligibleUsers[i]
+    const { referrerId, userAddress, timestamp } = eligibleUsers[i]
     console.log(
       `Calculating revenue for ${userAddress} (${i + 1}/${eligibleUsers.length})`,
     )
+
+    const referralDate = new Date(timestamp - REFERRAL_TIME_BUFFER_IN_SECONDS)
```

Comment:
This might be overkill/unnecessary, but is it worth skipping users whose `referralDate > endTimestamp`? It's possible(?) that if a user like this sneaks in, things will break/throw within the revenue calculation since `startTimestamp > endTimestamp` in that case.

Reply:
yes it's possible since we'll run the calculations for a past time period, i overlooked this scenario! thanks, updated :D